app/code/Magento/InventoryCatalog/Model/StockSourceLink/Validator/AssignToDefaultStockValidator.php

```diff
@@ -14,7 +14,7 @@
 use Magento\InventoryCatalog\Api\DefaultSourceProviderInterface;
 use Magento\InventoryCatalog\Api\DefaultStockProviderInterface;
 
-class AssignNonDefaultSourceToDefaultStockValidator implements StockSourceLinkValidatorInterface
```

Comment:
Looks like we can merge
`\Magento\InventoryCatalog\Model\StockSourceLink\Validator\AssignToDefaultStockValidator`
`\Magento\InventoryCatalog\Model\StockSourceLink\Validator\AssignDefaultSourceToNonDefaultStockValidator`

With condition
```
if ($link->getSourceCode() === $this->defaultSourceProvider->getCode() 
    || $link->getStockId() === $this->defaultStockProvider->getId()
) {
// coressponding message
}
```

Reply:
Merged.